service_container.rst

```diff
@@ -815,20 +815,35 @@ Public Versus Private Services
 From Symfony 4.0, every service defined is private by default.
 
 What does this mean? When a service **is** public, you can access it directly
-from the container object, which is accessible from any controller that extends
-``Controller``::
+from the container object, which can also be injected thanks to autowiring.
+This is mostly useful when you want to fetch services lazily::
 
-    use App\Service\MessageGenerator;
+    namespace App\Service;
```

Comment:
App\Twig

?

Reply:
No, this is not a twig extension we're building here, but a service that uses Twig lazily

Reply:
Hmm I don’t like the “Service” namespace much.

What about App\Generator\MessageGenerator?

Reply:
Done